avocado/utils/network/interfaces.py

```diff
@@ -313,6 +321,35 @@ def save(self, ipaddr, netmask):
                       'IPV6_DEFROUTE': 'yes'}
         if current_distro.name == 'SuSE':
             ifcfg_dict.pop('BOOTPROTO')
+
+        if self.if_type == 'Bond':
+            # Update master and slave configuration for bond
+            ifcfg_dict['BONDING_MASTER'] = 'yes'
+            mode, slaves = self._get_bondinterface_details().splitlines()
+            ifcfg_slave_dict = {'SLAVE': 'yes',
+                                'ONBOOT': 'yes',
+                                'MASTER': self.name}
+            if current_distro.name == 'SuSE':
+                ifcfg_dict['BONDING_MODULE_OPTS'] = 'mode='+mode.split()[0]
+                for j in range(0, len(slaves.split())):
```

Comment:
Your _get_bondinterface_details(), could return a list on slaves instead of a string, so here would be easier for you:

```python
for slave in slaves:
```